mobify/platform-scaffold
# Add placeholder children to navigation state

A change to where from the Redux store the home page's category list of links comes from broke the rendering of their placeholders. This change aims to fix it so placeholders still render.

## Changes
- Remove old placeholder array from the `UI.home` store
- Add new placeholder children to the `navigation` store
- Homepage category list links should now render placeholders again!

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Verify that between the Preloader and the fully loaded homepage, there is no time when the category list is not rendered (even if their content is incomplete)

## Result

| Before | After |
| --- | --- |
| Notice in this example the distinct lack of placeholder categories. | Notice how as soon as the homepage renders, the category list shows with placeholder content.
| ![before](https://user-images.githubusercontent.com/734535/27238412-9c1d3b44-5281-11e7-8528-837f5b506b09.gif) | ![after](https://user-images.githubusercontent.com/734535/27238418-a0ace10a-5281-11e7-9d55-a49939e6e2d9.gif) |
